### PR TITLE
Add high-performance datasets model

### PR DESCRIPTION
**Pull request by ThomasKroes**
To improve the `DatasetPickerAction`

Note: `DatasetPickerAction::setDatasetsFilterFunction(...)` has been renamed to `DatasetPickerAction::setFilterFunction(...)` where the filter function now has the following signature: `std::function<bool(Dataset<DatasetImpl>)>`.

Works in concert with the corresponding feature branch in the [scatterplot](https://github.com/ManiVaultStudio/Scatterplot/tree/feature/improve_dataset_picker_action_performance) plugin

### CODE REVIEW

File: ManiVault/src/actions/DatasetPickerAction.cpp [hunk 1]
```diff
@@ -48,35 +49,64 @@ DatasetPickerAction::DatasetPickerAction(QObject* parent, const QString& title)
...
     qDebug() << __FUNCTION__;
 #endif
 
     setPopulationMode(AbstractDatasetsModel::PopulationMode::Manual);
 
-    _datasetsListModel.setDatasets(datasets);
+    if (silent) {
+        blockDatasetsChangedSignal();
+        {
+            _datasetsListModel.setDatasets(datasets);
```
Comment (alxvth) [inline ManiVault/src/actions/DatasetPickerAction.cpp:119]: This only sets the new datasets if `silent` is `true`, right? Shouldn't it rather always set the datasets while `silent` only affects the signal?

Comment (ThomasKroes) [reply ManiVault/src/actions/DatasetPickerAction.cpp:119]: You are right, that slipped through... I'll fix it

Comment (alxvth) [reply ManiVault/src/actions/DatasetPickerAction.cpp:119]: Fixed!

**Comment by alxvth** [comment]
Works well 👍🏻

**Pull request merged**